sway/tree/container.c

```diff
@@ -433,6 +445,9 @@ void container_descendants(struct sway_container *root,
 			func(item, data);
 		}
 		container_descendants(item, type, func, data);
+		if (i < root->children->length && root->children->items[i] != item) {
```

Comment:
Hmm, this is a bit hacky, and only works if the current container is removed. I'd rather copy the list when doing this sort of thing.

Reply:
>I would prefer the PR be changed to use a double/nested for loop to iterate the workspaces

Well, that doesn't change the fact that the inner loop will iterate over a list that will be mutated. But I agree it's a better idea.

Reply:
I switched it to using a nested for loop. Since it's only ever removes the current element, it should be fine.